pennylane/ops/functions/matrix.py

```diff
@@ -0,0 +1,142 @@
+        contain multiple operations.
+
+        Consider the following quantum function:
+
+        .. code-block:: python3
+
+            def circuit(theta):
+                qml.RX(theta, wires=1)
+                qml.PauliZ(wires=0)
+
+        We can use ``qml.matrix`` to generate a new function that returns the unitary matrix
+        corresponding to the function ``circuit``:
+
+        >>> get_matrix = qml.matrix(circuit)
+        >>> theta = np.pi / 4
+        >>> get_matrix(theta)
+        array([[ 0.92387953+0.j,  0.+0.j ,  0.-0.38268343j,  0.+0.j],
+        [ 0.+0.j,  -0.92387953+0.j,  0.+0.j,  0. +0.38268343j],
+        [ 0. -0.38268343j,  0.+0.j,  0.92387953+0.j,  0.+0.j],
+        [ 0.+0.j,  0.+0.38268343j,  0.+0.j,  -0.92387953+0.j]])
+
+        Note that since ``wire_order`` was not specified, the default order ``[1, 0]`` for ``circuit``
+        was used, and the unitary matrix corresponds to the operation :math:`Z\otimes R_X(\theta)`. To
+        obtain the matrix for :math:`R_X(\theta)\otimes Z`, specify ``wire_order=[0, 1]`` in the
+        function call:
+
+        >>> get_matrix = qml.matrix(circuit, wire_order=[0, 1])
+
+        You can also get the unitary matrix for operations on a subspace of a larger Hilbert space. For
+        example, with the same function ``circuit`` and ``wire_order=["a", 0, "b", 1]`` you obtain the
+        :math:`16\times 16` matrix for the operation :math:`I\otimes Z\otimes I\otimes  R_X(\theta)`.
+
+        This unitary matrix can also be used in differentiable calculations. For example, consider the
+        following cost function:
+
+        .. code-block:: python
+
+            def circuit(theta):
+                qml.RX(theta, wires=1) qml.PauliZ(wires=0)
+                qml.CNOT(wires=[0, 1])
+
+            def cost(theta):
+                matrix = qml.matrix(circuit)(theta)
+                return np.real(np.trace(matrix))
+
+        Since this cost function returns a real scalar as a function of ``theta``, we can differentiate
+        it:
+
+        >>> theta = np.array(0.3, requires_grad=True)
+        >>> cost(theta)
+        1.9775421558720845
+        >>> qml.grad(cost)(theta)
+        -0.14943813247359922
+    """
+    if isinstance(op, qml.Hamiltonian):
+        return qml.utils.sparse_hamiltonian(op, wires=wire_order).toarray()
+
+    return op.get_matrix(wire_order=wire_order)
+
+
+@matrix.tape_transform
+def _matrix(tape, wire_order=None):
+    """Defines how matrix works if applied to a tape containing multiple operations."""
+    params = tape.get_parameters(trainable_only=False)
+    interface = qml.math._multi_dispatch(params)
+
+    wire_order = wire_order or tape.wires
+    n_wires = len(wire_order)
+
+    # initialize the unitary matrix
+    unitary_matrix = qml.math.eye(2**n_wires, like=interface)
+
+    for op in tape.operations:
+        U = matrix(op, wire_order=wire_order)
+        unitary_matrix = qml.math.dot(U, unitary_matrix)
+
+    return unitary_matrix

```

Comment:
Is this logic what's left from `get_unitary_matrix`?

Reply:
Yep 🙂 But much simplified